packages/sanity/src/adapter.ts

```
@@ -0,0 +1,159 @@
+                ...user,
+              })
+            })().then(
+              () => {},
+              () => {}
+            )
+
+            return cachedUser
+          }
+
+          const user = await client.fetch(getUserByIdQuery, {
+            id,
+          })
+
+          return {
+            id: user._id,
+            ...user,
+          }
+        },
+
+        async linkAccount(
+          userId,
+          providerId,
+          providerType,
+          providerAccountId,
+          refreshToken,
+          accessToken,
+          accessTokenExpires
+        ) {
+          await client.create({
+            _type: "account",
+            providerId,
+            providerType,
+            providerAccountId: `${providerAccountId}`,
+            refreshToken,
+            accessToken,
+            accessTokenExpires,
+            user: {
+              _type: "reference",
+              _ref: userId,
+            },
+          })
+        },
+
+        async getUserByProviderAccountId(providerId, providerAccountId) {
+          const account = await client.fetch(getUserByProviderAccountIdQuery, {
+            providerId,
+            providerAccountId: String(providerAccountId),
+          })
+
+          return account?.user
+        },
+
+        async getUserByEmail(email) {
+          const user = await client.fetch(getUserByEmailQuery, {
+            email,
+          })
+
+          return user
+        },
+
+        async createSession() {
+          console.log("[createSession] method not implemented")
+
+          return {} as any
+        },
+        async getSession() {
+          console.log("[getSession] method not implemented")
+          return {} as any
+        },
+        async updateSession() {
+          console.log("[updateSession] method not implemented")
+          return {} as any
+        },
+        async deleteSession() {
+          console.log("[deleteSession] method not implemented")
+        },
```

Just a question, do you have any plans on implementing these? If not, maybe we should throw an Error that tells the users that they have to use `jwt: true` for sessions.

Either way, I think `logger.error` would be a better option to log these, instead of `console.log`. You can get the `logger` from `appOptions` from the `getAdapter` parameter

---

@Fedeya if we don't implement these methods, maybe we should give a warning in the `README.md`, that this adapter only works when `session: { jwt: true }`, so people won't get surprised. Maybe even better, check `appOptions.session.jwt === true` at the top of `getAdapter(appOptions)`, and use `logger.warn` to give a heads up?